out_newrelic.go

```diff
@@ -53,6 +56,8 @@ func FLBPluginFlush(data unsafe.Pointer, length C.int, tag *C.char) int {
 		buffer = append(buffer, record.RemapRecord(fbRecord, ts, VERSION))
 	}
 
+	id := output.FLBPluginGetContext(ctx).(string)
```

Comment:
Hmmm is this actually working? You're actually doing a type assertion here, which casts the context into a string. But the context is not a string, so it will panic. I think you should follow a similar approach to what is done in the `FLBPluginInit` method. This will force you to parse the configuration again, but a this point you're not interested on the ProxyConfig parsing, for instance, so we might consider splitting the configuration parsing... :-/

Reply:
As we talked offline, it's the suggestion from the plugin maintainer. https://github.com/fluent/fluent-bit-go/blob/master/examples/out_multiinstance/out.go